source-container-build/app/source_build.py

```diff
@@ -381,23 +381,45 @@ def build_and_push(
     # push to registry
     fd, digest_file = tempfile.mkstemp()
     os.close(fd)
-    push_cmd = [
-        "skopeo",
-        "copy",
-        "--digestfile",
-        digest_file,
-        f"oci://{image_output_dir}:latest-source",
-        f"docker://{dest_image}",
-    ]
-    log.debug("push source image %r", push_cmd)
-    run(push_cmd, check=True)
+    for dest_image in dest_images:
+        push_cmd = [
+            "skopeo",
+            "copy",
+            "--digestfile",
+            digest_file,
+            f"oci://{image_output_dir}:latest-source",
+            f"docker://{dest_image}",
+        ]
+        log.debug("push source image %r", push_cmd)
+        run(push_cmd, check=True)
     try:
         with open(digest_file, "r") as f:
             build_result["image_digest"] = f.read().strip()
     finally:
         os.unlink(digest_file)
 
 
+def fetch_image_manifest_digest(image: str) -> str:
+    cmd = ["skopeo", "inspect", "--format", "{{.Digest}}", f"docker://{image}"]
+    return run(cmd, check=True, text=True, capture_output=True).stdout.strip()
```

Comment:
The default `skopeo inspect` can be very slow. `--no-tags` can help a bit:
```bash
$ time skopeo inspect --format '{{.Digest}}' docker://registry.access.redhat.com/ubi9/ubi-minimal
sha256:a340f4b9fb261a75c84666a3dccb88e193a116da3cebabaf9bcdc33609b61172
skopeo inspect --format '{{.Digest}}'   0.07s user 0.02s system 5% cpu 1.904 total

$ time skopeo inspect --format '{{.Digest}}' --no-tags docker://registry.access.redhat.com/ubi9/ubi-minimal
sha256:a340f4b9fb261a75c84666a3dccb88e193a116da3cebabaf9bcdc33609b61172
skopeo inspect --format '{{.Digest}}' --no-tags   0.07s user 0.02s system 7% cpu 1.256 total
```

And `--raw` is much faster:

```bash
$ time skopeo inspect --raw docker://registry.access.redhat.com/ubi9/ubi-minimal | sha256sum
a340f4b9fb261a75c84666a3dccb88e193a116da3cebabaf9bcdc33609b61172  -
skopeo inspect --raw docker://registry.access.redhat.com/ubi9/ubi-minimal  0.06s user 0.01s system 27% cpu 0.276 total
sha256sum  0.00s user 0.00s system 0% cpu 0.276 total
```

Reply:
Update the command with `--no-tags` and still rely on skopeo to calculate the digest.